### PR TITLE
Feature #78 - Filters record manager

### LINKED ISSUE #78: Als record manager wil ik mijn overzichtspagina VL kunnen ordenen adhv status van VL,
zodat ik bijvoorbeeld alleen de afgehandelde de lijsten kan tonen op mijn scherm (zoals de proceseigenaar of archivaris dit kunnen)

[overig - usability]

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #78 

Screenshot from 2021-04-08 17-29-37

### CODE REVIEW

File: src/archiefvernietigingscomponent/destruction/filters.py [hunk 1]
```diff
@@ -34,3 +35,40 @@ def filter_reviewed(self, queryset, name, value):
...
+        if not data.get("list_status"):
+            data["list_status"] = RecordManagerDisplay.all
+
+        super().__init__(data, *args, **kwargs)
+
+    def filter_list_status(self, queryset, name, value):
+        if value == RecordManagerDisplay.in_progress:
+            return queryset.filter(
+                Q(status=ListStatus.in_progress) | Q(status=ListStatus.processing),
+                ~Q(author=F("assignee")) | Q(assignee__isnull=True),
```
Comment (annashamray) [inline src/archiefvernietigingscomponent/destruction/filters.py:65]:
```suggestion
                Q(~Q(author=F("assignee"), status=ListStatus.in_progress) | Q(status=ListStatus.processing),
```
I'm not sure but I thought that `Q(assignee__isnull=True)` is a special case of `~Q(author=F("assignee"))` and is excessive
And I know that theoretically, an author can't be an assignee if `status=ListStatus.processing`, but still this case should be included in this `if`

**Review by annashamray — CHANGES_REQUESTED** [review]
DL filtering looks very nice! 

But I have some remarks about email preferences implementation since I haven't reviewed them. if you want they can be dealt with in a separate issue/PR.
* if the record-manager doesn't have email preferences configured, then the destruction-list page will return 500 error. The same applies to a reviewer list page. I think data migration is needed here. 
* `EmailPreference` model is not included into `admin_index` fixture
* it would be nice to have email preferences as inlines\field in the detail page and included into the `list_display` in the User admin (I don't quite understand why email preferences are put into a separate model. tbh)

**Comment by SilviaAmAm** [comment]
@annashamray I shall address the issues with the email preferences in another PR! :) But to clarify:
- Currently there is a signal that whenever a new user is created, it adds email preferences for that user. So any new user should have the email preferences. Maybe we should have a management command that checks for any old users without email preferences and creates them in case they are missing?
- I shall add it!
- Would it be better to add an email preference field to the user instead of having a separate model? Initially I thought it would be more flexible in case additional email preferences need to be added in the future, but in the end if it's just a single choice field I agree that it is maybe a bit overkill :P

**Comment by annashamray** [comment]
>     * Currently there is a signal that whenever a new user is created, it adds email preferences for that user. So any new user should have the email preferences. Maybe we should have a management command that checks for any old users without email preferences and creates them in case they are missing?

I think data migration should be enough. It should be run only once and if it's a migration it won't depend on humans :)

**Review by annashamray — APPROVED** [review]

**Pull request merged**